nixos/modules/system/boot/loader/generic-extlinux-compatible/extlinux-conf-builder.sh

```diff
@@ -6,16 +6,17 @@ export PATH=/empty
 for i in @path@; do PATH=$PATH:$i/bin; done
 
 usage() {
-    echo "usage: $0 -t <timeout> -c <path-to-default-configuration> [-d <boot-dir>] [-g <num-generations>] [-n <dtbName>] [-r]" >&2
+    echo "usage: $0 -t <timeout> [-p] -c <path-to-default-configuration> [-d <boot-dir>] [-g <num-generations>] [-n <dtbName>] [-r]" >&2
     exit 1
 }
 
 timeout=                # Timeout in centiseconds
+prompt=1                # Enable prompt
```

Comment:
```suggestion
prompt=0                # Enable prompt
```

Reply:
Hijacking this to document what things say about this... And it's a mess.

* * *

From syslinux `txt/syslinux.cfg.txt`


```
*PROMPT* 'flag_val'::                                                 
If 'flag_val' is 0, display the boot: prompt only if the Shift or Alt
key is pressed, or Caps Lock or Scroll lock is set (this is the   
default).  If 'flag_val' is 1, always display the boot: prompt.
```

* * *

From U-Boot `common/menu.c`

```
timeout - A delay in seconds to wait for user input. If 0, timeout is
disabled, and the default choice will be returned unless prompt is 1.

prompt - If 0, don't ask for user input unless there is an interrupted    
timeout. If 1, the user will be prompted for input regardless of the value
of timeout.                                                               
```

Side-note, in U-Boot, the default is `0` unless `MENU` is present in the config, which will set `prompt = 1`. So in practice since we have `MENU` it should set `prompt = 1`.

Though this all should be taken with a big pile of salt, since [the commit referenced](https://source.denx.de/u-boot/u-boot/-/commit/739e8361f3fe78038251216df6096a32bc2d5839) implies the documented behaviour might not have been matching reality. It is unclear from the commit alone if this was from a regression, or it never worked as documented.

* * *

# :melting_face: